tools/scripts/script.ps1

```diff
@@ -0,0 +1,230 @@
+# ----------------------------------------------------------------------------------
+#
+# Copyright Microsoft Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+param(
+    [Parameter(Mandatory = $false, Position = 0, HelpMessage = "To uninstall. By default false.")]
+    [switch]$uninstall
+)
+
+<#################################################
+#
+#               Helper functions
+#
+#################################################>
+
+function Get-PowerShellVersion{
+    if ($PSVersionTable.PSVersion.Major -lt 7){
+        Write-Warning "[For the best experience please install PowerShell version 7.]`n"
+    }
+}
+
+function Save-AIShellFromUrl{
+    # using placeholder urls until the GitHub release files are ready
+    $osxArm64Url = "osxArm64Url"
+    $osxX64Url = "osxX64Url"
+    $winArm64Url = "winArm64Url"
+    $winX64Url = "winX64Url"
+    $moduleUrl = "moduleUrl"
+
+    if (!$moduleUrl) {
+        return
+    }
+    
+    $destinationDirectory = (Join-Path (Join-Path (Join-Path $env:userprofile ".azure") "bin") "aishell")
+
+    # Create the directory if not existed
+    if(!(Test-Path -path $destinationDirectory))  {
+        New-Item -Path $destinationDirectory -ItemType Directory -Force | Out-Null
+    }
+
+    # Download AIShell Module
+    $destinationZipFile = Join-Path $destinationDirectory "AIShell-Module.zip"
+    Write-Host "[Downloading the latest AIShell PowerShell Module to $destinationZipFile ...]`n" -ForegroundColor Green
+    # Fix for where downloading with Invoke-WebRequest is too slow
+    # $ProgressPreference = 'SilentlyContinue'
+    Invoke-WebRequest -Uri $moduleUrl -OutFile $destinationZipFile
+    
+    # Unzip AIShell Module
+    $destinationUnzippedFolder = Join-Path $destinationDirectory "AIShell-Module"
+    Write-Host "[Unzipping the module to $destinationUnzippedFolder ...]`n" -ForegroundColor Green
+    # adding -ErrorAction SilentlyContinue because if user is not using Powershell as an administrator, the AIShell module dll files would not be able to be deleted until the current session has been closed. 
+    Expand-Archive -Path $destinationZipFile -DestinationPath $destinationUnzippedFolder -Force -ErrorAction SilentlyContinue
```

Comment:
Please make sure you unblock the zip file on Windows before extract it. You can use `Unblock-File` for that.